src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
       last.unsafeRunSync shouldBe LastTransactionRef("", 0L)
     }
 
-    "should return last transaction ref if address is known" in {}
+    "should return last transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+
+      val last = service.getLastTransactionRef(address).unsafeRunSync
+      last shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionRef" - {
+    "should return empty string and 0 ordinal if address is unknown" in {
+      val lastAccepted = service.getLastAcceptedTransactionRef("unknown").unsafeRunSync
+
+      lastAccepted shouldEqual LastTransactionRef("", 0L)
+    }
+
+    "should return last accepted transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastAcceptedTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+      val lastAccepted = service.getLastAcceptedTransactionRef(address).unsafeRunSync
+
+      lastAccepted shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionMap" - {
+    "should return current map of accepted transactions" in {
+      service.getLastAcceptedTransactionMap().unsafeRunSync shouldEqual Map.empty
+
+      val lastAcceptedTxRef = Map("DAG123" -> LastTransactionRef("abc", 1))
+      service.lastAcceptedTransactionRef.set(lastAcceptedTxRef).unsafeRunSync
+      val result = service.getLastAcceptedTransactionMap().unsafeRunSync
+
+      result shouldEqual lastAcceptedTxRef
+    }
+  }
+
+  "acceptTransaction" - {
+    "for dummy transaction" - {
+      "should not modify lastAcceptedTransactionRef" in {
+        val dummyTxEdge = mock[Edge[TransactionEdgeData]]
+        val dummyTx = Transaction(dummyTxEdge, LastTransactionRef.empty, isDummy = true)
+        service.acceptTransaction(dummyTx).unsafeRunSync
+
+        val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+
+        result shouldEqual Map.empty
+      }
+    }
+    "for normal transaction" - {
+      "should add transaction to lastAcceptedTransactionRef" in {
```

Comment:
maybe without using internal names, like:
```
for dummy - should not persist transaction ref
for normal - should persist transaction ref
```
wdyt?

Reply:
I aggree, will fix that!